File: scripts/generate-icons.py
"""
NewShare App Icon Generator using Pillow
Creates all required icon PNG files
"""
from PIL import Image, ImageDraw, ImageFilter
import math
import os

ASSETS_DIR = os.path.join(os.path.dirname(__file__), "../assets/images")

# Color palette
DARK_BG1 = (26, 26, 46)      # #1a1a2e
DARK_BG2 = (22, 33, 62)      # #16213e
ACCENT1  = (233, 69, 96)     # #e94560 (red-pink)
ACCENT2  = (245, 166, 35)    # #f5a623 (orange)
WHITE    = (255, 255, 255)
DEEP_BLUE = (15, 52, 96)     # #0f3460


def lerp_color(c1, c2, t):
    return tuple(int(c1[i] + (c2[i] - c1[i]) * t) for i in range(3))


def make_gradient_bg(size, c1=DARK_BG1, c2=DARK_BG2):
    img = Image.new("RGBA", (size, size), (0, 0, 0, 0))
    draw = ImageDraw.Draw(img)
    for y in range(size):
        t = y / size
        # diagonal gradient
        color = lerp_color(c1, c2, t)
        draw.line([(0, y), (size, y)], fill=color + (255,))
    return img


def rounded_rect_mask(size, radius_ratio=0.215):
    """Create a rounded rectangle mask."""
    mask = Image.new("L", (size, size), 0)
    draw = ImageDraw.Draw(mask)
    r = int(size * radius_ratio)
    draw.rounded_rectangle([0, 0, size - 1, size - 1], radius=r, fill=255)
    return mask


def draw_N(draw, size, gradient=True):
    """Draw the N letterform with gradient fill."""
    s = size / 1024  # scale factor

    bar_w = int(110 * s)
    bar_h = int(444 * s)
    top_y = int(290 * s)
    left_x = int(248 * s)
    right_x = int(666 * s)

    # Draw gradient N by drawing horizontal scanlines
    # Left vertical bar
    for y in range(bar_h):
        t = y / bar_h
        color = lerp_color(ACCENT1, ACCENT2, t)
        draw.rectangle([left_x, top_y + y, left_x + bar_w, top_y + y], fill=color + (255,))

    # Right vertical bar
    for y in range(bar_h):
        t = y / bar_h
        color = lerp_color(ACCENT1, ACCENT2, t)
        draw.rectangle([right_x, top_y + y, right_x + bar_w, top_y + y], fill=color + (255,))

    # Diagonal stroke: polygon from (248,290) (358,290) (776,734) (666,734)
    pts = [
        (int(248 * s), int(290 * s)),
        (int(358 * s), int(290 * s)),
        (int(776 * s), int(734 * s)),
        (int(666 * s), int(734 * s)),
    ]
    # Draw diagonal with gradient scanlines (approximate)
    for y in range(bar_h):
        t = y / bar_h
        color = lerp_color(ACCENT1, ACCENT2, t)
        y_coord = top_y + y
        # Find x bounds of diagonal at this y
        # Left edge: line from (248,290) to (666,734) → slope (666-248)/(734-290) = 418/444
        # Right edge: line from (358,290) to (776,734)
        dy = y_coord - int(290 * s)
        total_dy = int(444 * s)
        left_edge = int(248 * s) + int(dy * 418 * s / (444 * s))
        right_edge = int(358 * s) + int(dy * 418 * s / (444 * s))
        if right_edge > left_edge:
            draw.rectangle([left_edge, y_coord, right_edge, y_coord], fill=color + (255,))


def draw_share_dots(draw, size):
    """Draw the share/broadcast dots in top-right area."""
    s = size / 1024

    cx = int(680 * s)
    cy = int(240 * s)

    r_main = int(28 * s)
    r_small = int(20 * s)
    offset_x = int(72 * s)
    offset_y = int(52 * s)

    line_w = max(1, int(6 * s))

    # Center dot (white)
    draw.ellipse([cx - r_main, cy - r_main, cx + r_main, cy + r_main],
                 fill=WHITE + (242,))

    # Top-left dot
    x1, y1 = cx - offset_x, cy - offset_y
    draw.ellipse([x1 - r_small, y1 - r_small, x1 + r_small, y1 + r_small],
                 fill=WHITE + (178,))

    # Bottom-left dot
    x2, y2 = cx - offset_x, cy + offset_y
    draw.ellipse([x2 - r_small, y2 - r_small, x2 + r_small, y2 + r_small],
                 fill=WHITE + (178,))

    # Connecting lines
    draw.line([x1, y1, cx, cy], fill=WHITE + (150,), width=line_w)
    draw.line([x2, y2, cx, cy], fill=WHITE + (150,), width=line_w)


def generate_main_icon(size=1024):
    img = make_gradient_bg(size)
    mask = rounded_rect_mask(size)

    # Accent circles
    draw = ImageDraw.Draw(img)
    s = size / 1024
    # Top-right circle
    cx, cy, r = int(820 * s), int(200 * s), int(180 * s)
    circle_img = Image.new("RGBA", (size, size), (0, 0, 0, 0))
    cdraw = ImageDraw.Draw(circle_img)
    cdraw.ellipse([cx - r, cy - r, cx + r, cy + r], fill=DEEP_BLUE + (38,))
    img = Image.alpha_composite(img, circle_img)

    # Bottom-left circle
    circle_img2 = Image.new("RGBA", (size, size), (0, 0, 0, 0))
    cdraw2 = ImageDraw.Draw(circle_img2)
    cx2, cy2, r2 = int(200 * s), int(820 * s), int(140 * s)
    cdraw2.ellipse([cx2 - r2, cy2 - r2, cx2 + r2, cy2 + r2], fill=DEEP_BLUE + (25,))
    img = Image.alpha_composite(img, circle_img2)

    # Draw N and dots on a separate layer for glow effect
    n_layer = Image.new("RGBA", (size, size), (0, 0, 0, 0))
    n_draw = ImageDraw.Draw(n_layer)
    draw_N(n_draw, size)
    draw_share_dots(n_draw, size)

    # Apply subtle glow
    glow = n_layer.filter(ImageFilter.GaussianBlur(int(8 * s)))
    img = Image.alpha_composite(img, glow)
    img = Image.alpha_composite(img, n_layer)

    # Apply rounded corners
    result = Image.new("RGBA", (size, size), (0, 0, 0, 0))
    result.paste(img, mask=mask)
    return result


def generate_android_foreground(size=1024):
    img = Image.new("RGBA", (size, size), (0, 0, 0, 0))
    s = size / 1024
    scale = 0.85

    # Draw scaled N
    n_layer = Image.new("RGBA", (size, size), (0, 0, 0, 0))
    n_draw = ImageDraw.Draw(n_layer)

    bar_w = int(110 * s * scale)
    bar_h = int(444 * s * scale)
    offset_x = int((1 - scale) * 512 * s)
    offset_y = int((1 - scale) * 512 * s)
    top_y = int(290 * s * scale) + offset_y
    left_x = int(248 * s * scale) + offset_x
    right_x = int(666 * s * scale) + offset_x

    for y in range(bar_h):
        t = y / bar_h
        color = lerp_color(ACCENT1, ACCENT2, t)
        n_draw.rectangle([left_x, top_y + y, left_x + bar_w, top_y + y], fill=color + (255,))
        n_draw.rectangle([right_x, top_y + y, right_x + bar_w, top_y + y], fill=color + (255,))
        dy = y
        left_edge = left_x + int(dy * 418 * s * scale / (444 * s * scale))
        right_edge = left_x + bar_w + int(dy * 418 * s * scale / (444 * s * scale))
        n_draw.rectangle([left_edge, top_y + y, right_edge, top_y + y], fill=color + (255,))

    # Share dots (adjusted position)
    cx = int(680 * s * scale) + offset_x
    cy = int(240 * s * scale) + offset_y
    r_main = int(28 * s)
    r_small = int(20 * s)
    ox = int(72 * s)
    oy = int(52 * s)
    lw = max(1, int(6 * s))
    n_draw.ellipse([cx - r_main, cy - r_main, cx + r_main, cy + r_main], fill=WHITE + (242,))
    n_draw.ellipse([cx - ox - r_small, cy - oy - r_small, cx - ox + r_small, cy - oy + r_small], fill=WHITE + (178,))
    n_draw.ellipse([cx - ox - r_small, cy + oy - r_small, cx - ox + r_small, cy + oy + r_small], fill=WHITE + (178,))
    n_draw.line([cx - ox, cy - oy, cx, cy], fill=WHITE + (150,), width=lw)
    n_draw.line([cx - ox, cy + oy, cx, cy], fill=WHITE + (150,), width=lw)

    glow = n_layer.filter(ImageFilter.GaussianBlur(6))
    img = Image.alpha_composite(img, glow)
    img = Image.alpha_composite(img, n_layer)
    return img


def generate_android_background(size=1024):
    img = make_gradient_bg(size)
    draw = ImageDraw.Draw(img)
    s = size / 1024
    # Decorative circles
    circle_img = Image.new("RGBA", (size, size), (0, 0, 0, 0))
    cdraw = ImageDraw.Draw(circle_img)
    cx, cy, r = int(820 * s), int(200 * s), int(180 * s)
    cdraw.ellipse([cx - r, cy - r, cx + r, cy + r], fill=DEEP_BLUE + (76,))
    cx2, cy2, r2 = int(200 * s), int(820 * s), int(140 * s)
    cdraw.ellipse([cx2 - r2, cy2 - r2, cx2 + r2, cy2 + r2], fill=DEEP_BLUE + (51,))
    img = Image.alpha_composite(img, circle_img)
    # Convert to RGB for Android background (no transparency)
    result = Image.new("RGB", (size, size))
    result.paste(img, mask=img.split()[3])
    return result.convert("RGBA")


def generate_android_monochrome(size=1024):
    img = Image.new("RGBA", (size, size), (0, 0, 0, 0))
    draw = ImageDraw.Draw(img)
    s = size / 1024
    scale = 0.85
    offset = int((1 - scale) * 512 * s)

    bar_w = int(110 * s * scale)
    bar_h = int(444 * s * scale)
    top_y = int(290 * s * scale) + offset
    left_x = int(248 * s * scale) + offset
    right_x = int(666 * s * scale) + offset

    draw.rectangle([left_x, top_y, left_x + bar_w, top_y + bar_h], fill=WHITE + (255,))
    draw.rectangle([right_x, top_y, right_x + bar_w, top_y + bar_h], fill=WHITE + (255,))
    for y in range(bar_h):
        dy = y
        left_edge = left_x + int(dy * 418 * s * scale / (444 * s * scale))
        right_edge = left_x + bar_w + int(dy * 418 * s * scale / (444 * s * scale))
        draw.rectangle([left_edge, top_y + y, right_edge, top_y + y], fill=WHITE + (255,))

    cx = int(680 * s * scale) + offset
    cy = int(240 * s * scale) + offset
    r_main = int(28 * s)
    r_small = int(20 * s)
    ox, oy = int(72 * s), int(52 * s)
    lw = max(1, int(6 * s))
    draw.ellipse([cx - r_main, cy - r_main, cx + r_main, cy + r_main], fill=WHITE + (255,))
    draw.ellipse([cx - ox - r_small, cy - oy - r_small, cx - ox + r_small, cy - oy + r_small], fill=WHITE + (204,))
    draw.ellipse([cx - ox - r_small, cy + oy - r_small, cx - ox + r_small, cy + oy + r_small], fill=WHITE + (204,))
    draw.line([cx - ox, cy - oy, cx, cy], fill=WHITE + (200,), width=lw)
    draw.line([cx - ox, cy + oy, cx, cy], fill=WHITE + (200,), width=lw)
    return img


def generate_favicon(size=64):
    img = make_gradient_bg(size, DARK_BG1, DARK_BG2)
    mask = rounded_rect_mask(size, 0.22)
    draw = ImageDraw.Draw(img)
    s = size / 64

    bar_w = int(8 * s)
    bar_h = int(32 * s)
    top_y = int(16 * s)
    left_x = int(14 * s)
    right_x = int(42 * s)

    for y in range(bar_h):
        t = y / bar_h
        color = lerp_color(ACCENT1, ACCENT2, t)
        draw.rectangle([left_x, top_y + y, left_x + bar_w, top_y + y], fill=color + (255,))
        draw.rectangle([right_x, top_y + y, right_x + bar_w, top_y + y], fill=color + (255,))

    # Diagonal
    for y in range(bar_h):
        t = y / bar_h
        color = lerp_color(ACCENT1, ACCENT2, t)
        dy = y
        left_edge = left_x + int(dy * (right_x - left_x) / bar_h)
        right_edge = left_x + bar_w + int(dy * (right_x - left_x) / bar_h)
        draw.rectangle([left_edge, top_y + y, right_edge, top_y + y], fill=color + (255,))

    result = Image.new("RGBA", (size, size), (0, 0, 0, 0))
    result.paste(img, mask=mask)
    return result


def generate_splash_icon(size=512):
    img = Image.new("RGBA", (size, size), (0, 0, 0, 0))
    s = size / 512
    n_layer = Image.new("RGBA", (size, size), (0, 0, 0, 0))
    n_draw = ImageDraw.Draw(n_layer)

    bar_w = int(56 * s)
    bar_h = int(272 * s)
    top_y = int(120 * s)
    left_x = int(100 * s)
    right_x = int(356 * s)

    for y in range(bar_h):
        t = y / bar_h
        color = lerp_color(ACCENT1, ACCENT2, t)
        n_draw.rectangle([left_x, top_y + y, left_x + bar_w, top_y + y], fill=color + (255,))
        n_draw.rectangle([right_x, top_y + y, right_x + bar_w, top_y + y], fill=color + (255,))
        dy = y
        left_edge = left_x + int(dy * (right_x - left_x) / bar_h)
        right_edge = left_x + bar_w + int(dy * (right_x - left_x) / bar_h)
        n_draw.rectangle([left_edge, top_y + y, right_edge, top_y + y], fill=color + (255,))

    # Share dots
    cx, cy = int(360 * s), int(100 * s)
    r_main = int(16 * s)
    r_small = int(12 * s)
    ox, oy = int(40 * s), int(30 * s)
    lw = max(1, int(4 * s))
    n_draw.ellipse([cx - r_main, cy - r_main, cx + r_main, cy + r_main], fill=WHITE + (242,))
    n_draw.ellipse([cx - ox - r_small, cy - oy - r_small, cx - ox + r_small, cy - oy + r_small], fill=WHITE + (178,))
    n_draw.ellipse([cx - ox - r_small, cy + oy - r_small, cx - ox + r_small, cy + oy + r_small], fill=WHITE + (178,))
    n_draw.line([cx - ox, cy - oy, cx, cy], fill=WHITE + (150,), width=lw)
    n_draw.line([cx - ox, cy + oy, cx, cy], fill=WHITE + (150,), width=lw)

    glow = n_layer.filter(ImageFilter.GaussianBlur(4))
    img = Image.alpha_composite(img, glow)
    img = Image.alpha_composite(img, n_layer)
    return img


def save(img, filename):
    path = os.path.join(ASSETS_DIR, filename)
    img.save(path, "PNG")
    size = os.path.getsize(path)
    print(f"✓ {filename} ({size // 1024}KB)")


if __name__ == "__main__":
    print("Generating NewShare icons...\n")

    save(generate_main_icon(1024), "icon.png")
    save(generate_android_foreground(1024), "android-icon-foreground.png")
    save(generate_android_background(1024), "android-icon-background.png")
    save(generate_android_monochrome(1024), "android-icon-monochrome.png")
    save(generate_favicon(64), "favicon.png")
    save(generate_splash_icon(512), "splash-icon.png")

    print("\n✅ All icons generated!")

File: scripts/generate-icons.mjs
/**
 * NewShare App Icon Generator
 * Creates all required icon PNG files from SVG templates
 */
import { execSync } from "child_process";
import { writeFileSync, existsSync } from "fs";
import { join, dirname } from "path";
import { fileURLToPath } from "url";

const __dirname = dirname(fileURLToPath(import.meta.url));
const assetsDir = join(__dirname, "../assets/images");

// Main icon SVG - modern gradient design with N + share symbol
const mainIconSvg = `<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 1024 1024" width="1024" height="1024">
  <defs>
    <linearGradient id="bg" x1="0%" y1="0%" x2="100%" y2="100%">
      <stop offset="0%" style="stop-color:#1a1a2e"/>
      <stop offset="100%" style="stop-color:#16213e"/>
    </linearGradient>
    <linearGradient id="accent" x1="0%" y1="0%" x2="100%" y2="100%">
      <stop offset="0%" style="stop-color:#0f3460"/>
      <stop offset="100%" style="stop-color:#533483"/>
    </linearGradient>
    <linearGradient id="nGrad" x1="0%" y1="0%" x2="100%" y2="100%">
      <stop offset="0%" style="stop-color:#e94560"/>
      <stop offset="100%" style="stop-color:#f5a623"/>
    </linearGradient>
    <filter id="glow">
      <feGaussianBlur stdDeviation="8" result="coloredBlur"/>
      <feMerge>
        <feMergeNode in="coloredBlur"/>
        <feMergeNode in="SourceGraphic"/>
      </feMerge>
    </filter>
  </defs>

  <!-- Background -->
  <rect width="1024" height="1024" rx="220" fill="url(#bg)"/>

  <!-- Subtle inner glow ring -->
  <rect width="1024" height="1024" rx="220" fill="none" stroke="url(#accent)" stroke-width="2" opacity="0.4"/>

  <!-- Decorative circles -->
  <circle cx="820" cy="200" r="180" fill="url(#accent)" opacity="0.15"/>
  <circle cx="200" cy="820" r="140" fill="url(#accent)" opacity="0.1"/>

  <!-- N letterform - bold, modern -->
  <g filter="url(#glow)">
    <!-- Left vertical bar -->
    <rect x="248" y="290" width="110" height="444" rx="16" fill="url(#nGrad)"/>
    <!-- Right vertical bar -->
    <rect x="666" y="290" width="110" height="444" rx="16" fill="url(#nGrad)"/>
    <!-- Diagonal stroke -->
    <polygon points="248,290 358,290 776,734 666,734" fill="url(#nGrad)"/>
  </g>

  <!-- Share dots - top right of N -->
  <g transform="translate(680, 240)" filter="url(#glow)">
    <!-- Center dot -->
    <circle cx="0" cy="0" r="28" fill="#ffffff" opacity="0.95"/>
    <!-- Top-left dot -->
    <circle cx="-72" cy="-52" r="20" fill="#ffffff" opacity="0.7"/>
    <!-- Bottom-left dot -->
    <circle cx="-72" cy="52" r="20" fill="#ffffff" opacity="0.7"/>
    <!-- Connecting lines -->
    <line x1="-52" y1="-40" x2="-20" y2="-8" stroke="#ffffff" stroke-width="6" opacity="0.6" stroke-linecap="round"/>
    <line x1="-52" y1="40" x2="-20" y2="8" stroke="#ffffff" stroke-width="6" opacity="0.6" stroke-linecap="round"/>
  </g>
</svg>`;

// Android foreground icon (slightly smaller content, transparent bg)
const androidForegroundSvg = `<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 1024 1024" width="1024" height="1024">
  <defs>
    <linearGradient id="nGrad" x1="0%" y1="0%" x2="100%" y2="100%">
      <stop offset="0%" style="stop-color:#e94560"/>
      <stop offset="100%" style="stop-color:#f5a623"/>
    </linearGradient>
    <filter id="glow">
      <feGaussianBlur stdDeviation="6" result="coloredBlur"/>
      <feMerge>
        <feMergeNode in="coloredBlur"/>
        <feMergeNode in="SourceGraphic"/>
      </feMerge>
    </filter>
  </defs>

  <!-- N letterform -->
  <g filter="url(#glow)" transform="translate(512,512) scale(0.85) translate(-512,-512)">
    <rect x="248" y="290" width="110" height="444" rx="16" fill="url(#nGrad)"/>
    <rect x="666" y="290" width="110" height="444" rx="16" fill="url(#nGrad)"/>
    <polygon points="248,290 358,290 776,734 666,734" fill="url(#nGrad)"/>
  </g>

  <!-- Share dots -->
  <g transform="translate(680, 240)" filter="url(#glow)">
    <circle cx="0" cy="0" r="28" fill="#ffffff" opacity="0.95"/>
    <circle cx="-72" cy="-52" r="20" fill="#ffffff" opacity="0.7"/>
    <circle cx="-72" cy="52" r="20" fill="#ffffff" opacity="0.7"/>
    <line x1="-52" y1="-40" x2="-20" y2="-8" stroke="#ffffff" stroke-width="6" opacity="0.6" stroke-linecap="round"/>
    <line x1="-52" y1="40" x2="-20" y2="8" stroke="#ffffff" stroke-width="6" opacity="0.6" stroke-linecap="round"/>
  </g>
</svg>`;

// Android background - solid dark blue
const androidBackgroundSvg = `<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 1024 1024" width="1024" height="1024">
  <defs>
    <linearGradient id="bg" x1="0%" y1="0%" x2="100%" y2="100%">
      <stop offset="0%" style="stop-color:#1a1a2e"/>
      <stop offset="100%" style="stop-color:#16213e"/>
    </linearGradient>
  </defs>
  <rect width="1024" height="1024" fill="url(#bg)"/>
  <circle cx="820" cy="200" r="180" fill="#0f3460" opacity="0.3"/>
  <circle cx="200" cy="820" r="140" fill="#0f3460" opacity="0.2"/>
</svg>`;

// Android monochrome (white on transparent)
const androidMonochromeSvg = `<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 1024 1024" width="1024" height="1024">
  <g transform="translate(512,512) scale(0.85) translate(-512,-512)">
    <rect x="248" y="290" width="110" height="444" rx="16" fill="#ffffff"/>
    <rect x="666" y="290" width="110" height="444" rx="16" fill="#ffffff"/>
    <polygon points="248,290 358,290 776,734 666,734" fill="#ffffff"/>
  </g>
  <g transform="translate(680, 240)">
    <circle cx="0" cy="0" r="28" fill="#ffffff"/>
    <circle cx="-72" cy="-52" r="20" fill="#ffffff" opacity="0.8"/>
    <circle cx="-72" cy="52" r="20" fill="#ffffff" opacity="0.8"/>
    <line x1="-52" y1="-40" x2="-20" y2="-8" stroke="#ffffff" stroke-width="6" stroke-linecap="round"/>
    <line x1="-52" y1="40" x2="-20" y2="8" stroke="#ffffff" stroke-width="6" stroke-linecap="round"/>
  </g>
</svg>`;

// Favicon SVG (simple, readable at small size)
const faviconSvg = `<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 64 64" width="64" height="64">
  <defs>
    <linearGradient id="bg" x1="0%" y1="0%" x2="100%" y2="100%">
      <stop offset="0%" style="stop-color:#1a1a2e"/>
      <stop offset="100%" style="stop-color:#16213e"/>
    </linearGradient>
    <linearGradient id="nGrad" x1="0%" y1="0%" x2="100%" y2="100%">
      <stop offset="0%" style="stop-color:#e94560"/>
      <stop offset="100%" style="stop-color:#f5a623"/>
    </linearGradient>
  </defs>
  <rect width="64" height="64" rx="14" fill="url(#bg)"/>
  <!-- N letterform (simplified for small size) -->
  <rect x="14" y="16" width="8" height="32" rx="2" fill="url(#nGrad)"/>
  <rect x="42" y="16" width="8" height="32" rx="2" fill="url(#nGrad)"/>
  <polygon points="14,16 22,16 50,48 42,48" fill="url(#nGrad)"/>
</svg>`;

// Splash icon SVG (centered, with extra padding)
const splashIconSvg = `<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512" width="512" height="512">
  <defs>
    <linearGradient id="nGrad" x1="0%" y1="0%" x2="100%" y2="100%">
      <stop offset="0%" style="stop-color:#e94560"/>
      <stop offset="100%" style="stop-color:#f5a623"/>
    </linearGradient>
    <filter id="glow">
      <feGaussianBlur stdDeviation="4" result="coloredBlur"/>
      <feMerge>
        <feMergeNode in="coloredBlur"/>
        <feMergeNode in="SourceGraphic"/>
      </feMerge>
    </filter>
  </defs>
  <!-- N letterform -->
  <g filter="url(#glow)">
    <rect x="100" y="120" width="56" height="272" rx="10" fill="url(#nGrad)"/>
    <rect x="356" y="120" width="56" height="272" rx="10" fill="url(#nGrad)"/>
    <polygon points="100,120 156,120 412,392 356,392" fill="url(#nGrad)"/>
  </g>
  <!-- Share dots -->
  <g transform="translate(360, 100)" filter="url(#glow)">
    <circle cx="0" cy="0" r="16" fill="#ffffff" opacity="0.95"/>
    <circle cx="-40" cy="-30" r="12" fill="#ffffff" opacity="0.7"/>
    <circle cx="-40" cy="30" r="12" fill="#ffffff" opacity="0.7"/>
    <line x1="-28" y1="-22" x2="-14" y2="-6" stroke="#ffffff" stroke-width="4" opacity="0.6" stroke-linecap="round"/>
    <line x1="-28" y1="22" x2="-14" y2="6" stroke="#ffffff" stroke-width="4" opacity="0.6" stroke-linecap="round"/>
  </g>
</svg>`;

// Write SVG files
const svgFiles = {
  "icon.svg": mainIconSvg,
  "android-icon-foreground.svg": androidForegroundSvg,
  "android-icon-background.svg": androidBackgroundSvg,
  "android-icon-monochrome.svg": androidMonochromeSvg,
  "favicon.svg": faviconSvg,
  "splash-icon.svg": splashIconSvg,
};

for (const [filename, content] of Object.entries(svgFiles)) {
  writeFileSync(join(assetsDir, filename), content);
  console.log(`✓ Written ${filename}`);
}

// Convert SVG to PNG using available tool
function convertSvgToPng(svgPath, pngPath, width, height) {
  // Try rsvg-convert first, then inkscape, then convert (ImageMagick)
  const commands = [
    `rsvg-convert -w ${width} -h ${height} "${svgPath}" -o "${pngPath}"`,
    `inkscape --export-type=png --export-width=${width} --export-height=${height} --export-filename="${pngPath}" "${svgPath}"`,
    `convert -background none -resize ${width}x${height} "${svgPath}" "${pngPath}"`,
    `magick -background none -resize ${width}x${height} "${svgPath}" "${pngPath}"`,
  ];

  for (const cmd of commands) {
    try {
      execSync(cmd, { stdio: "pipe" });
      console.log(`✓ Converted ${svgPath} -> ${pngPath} (${width}x${height})`);
      return true;
    } catch {
      // Try next command
    }
  }
  console.error(`✗ Could not convert ${svgPath} to PNG. Install rsvg-convert, inkscape, or imagemagick.`);
  return false;
}

const conversions = [
  ["icon.svg", "icon.png", 1024, 1024],
  ["android-icon-foreground.svg", "android-icon-foreground.png", 1024, 1024],
  ["android-icon-background.svg", "android-icon-background.png", 1024, 1024],
  ["android-icon-monochrome.svg", "android-icon-monochrome.png", 1024, 1024],
  ["favicon.svg", "favicon.png", 64, 64],
  ["splash-icon.svg", "splash-icon.png", 512, 512],
];

let allConverted = true;
for (const [svgFile, pngFile, w, h] of conversions) {
  const ok = convertSvgToPng(join(assetsDir, svgFile), join(assetsDir, pngFile), w, h);
  if (!ok) allConverted = false;
}

if (allConverted) {
  console.log("\n✅ All icons generated successfully!");
} else {
  console.log("\n⚠️  Some icons could not be converted. Install rsvg-convert: sudo apt-get install librsvg2-bin");
}
